### ISSUE TITLE
When using accelerate with deepspeed and peft, any accelerator_kwargs passed to PPOConfig( ) are not used?

### ISSUE THREAD

**Issue by sandeepchittilla** [lvwerra/trl#304]
**A bit of context:**

I am launching my script with `accelerate launch my_script.py` with a custom accelerate config file. I'm running this in a distributed training setting across 4 GPUs.

While saving a model, I get NCCL Timeout errors (default is 30mins). So, I increase the timeout to 1.5hr by passing `kwargs_handlers=[InitProcessGroupKwargs(timeout=timedelta(seconds=5_400))]` during Accelerator initialization in `ppo_trainer.py`.

However, these values are not used becauase the accelerator has already been initialized probably since I launch it via cli. 

Is there a workaround for this? Or an option to edit the state in case some previously unset parameter is passed? Or .... I'm missing something?

**Comment by LexLuter1** [comment]
asd

**Comment by lvwerra** [comment]
The kwargs issue seems to be an `accelerate` issue, would you mind raising it [there](https://github.com/huggingface/accelerate/issues)? 30min should be more than enough time to save the model, I suspect something else is off when you save the model. How do you save the model?

**Comment by sandeepchittilla** [comment]
@lvwerra sure I will do that. Yes, I felt that 30 mins is plenty of time for saving. I save the model using the `_save_pretrained` in `ppo_trainer` like so :

```python
if ppo_trainer.accelerator.is_main_process:
        if ppo_trainer.is_distributed:
            torch.distributed.barrier()
        os.makedirs(os.path.join(output_dir, f"checkpoint_{chkpt_name}"), exist_ok=True)
        ppo_trainer._save_pretrained(os.path.join(output_dir, f"checkpoint_{chkpt_name}"))
```

Currently, I am trying to save after every optimization step. I use :  Accelerate with deepspeed stage 2 on 4 gpus with gradient accumulation over 8 steps
